content/docs/2.6/scalers/azure-app-insights.md

```diff
@@ -0,0 +1,120 @@
++++
+title = "Azure Application Insights"
+layout = "scaler"
+availability = "v1.3+"
+maintainer = "Community"
+description = "Scale applications based on Azure Application Insights metrics."
+go_file = "azure_app_insights_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `azure-app-insights` trigger that scales based on an Azure Application Insights metric.
+
+```yaml
+triggers:
+- type: azure-app-insights
+  metadata:
+    metricAggregationTimespan: "0:1"
+    metricAggregationType: avg
+    metricFilter: cloud/roleName eq 'role_name'
+    metricName: "metric_name"
+    metricNamespace: "customMetrics"
```

Comment:
Do we need the metric namespace here?

Reply:
I don't think so as it doesn't match the API. I've removed the parameter and updated the examples.

Reply:
Actually I think my initial comment was wrong, sorry! Let me rephrase that: Do we need the `metric` prefix? Can't we just use `aggregationType`, `filter`, etc?

It would be good to keep in the `namespace` btw so that people can change it if they need to, but we can use `customMetrics` as a default.

Reply:
Some of this discussion is related to the discussion below around the metricName parameter. The REST API documented [here](https://dev.applicationinsights.io/reference) refers to the metric to be retrieved as the metricId. Some examples are:
- availabilityResults/availabilityPercentage
- browserTimings/networkDuration
- customMetrics/customMetricForSomeService

The suggestion below is to change metricName to metricId to be consistent with the API. If that's ok, that leaves us with the metricId, aggregationType, and filter fields. Those fields will allow us to do things like aggregate a metric using a given dimension.

Reply:
Sounds good to me!

Reply:
Great! Thanks. I'll get those changes in later today.